framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -566,177 +556,143 @@ protected void reportPurityErrors(
         }
     }
 
-    /**
-     * Checks all (non-conditional) postcondition on the method {@code node} with element {@code
-     * methodElement}.
-     */
-    protected void checkPostconditions(MethodTree node, ExecutableElement methodElement) {
+    protected void checkContractsAtMethodDeclaration(
```

Comment:
Javadoc for protected methods is very useful. Can yo add something for this method? When would a type system override it?